payment_source.go

```diff
@@ -0,0 +1,67 @@
+package stripe
+
+import (
+	"encoding/json"
+	"github.com/stripe/stripe-go/paymentsource"
+)
+
+// SourceParams is the set of parameters that can be used to describe the source
+// object used to make a Charge. For example, both Card and BitcoinReceiver objects
+// can be described by SourceParams.
+// For more details see https://stripe.com/docs/api#create_charge
+type SourceParams struct {
+	Params
+	ID    string
+	Token string
+	Card  *CardParams
+}
+
+// Interface to provide a human readable representation of a struct
+type Displayer interface {
+	Display() string
+}
+
+// PaymentSource describes the payment source used to make a Charge.
+// The Type should indicate which object is fleshed out (eg. BitcoinReceiver or Card)
+// For more details see https://stripe.com/docs/api#retrieve_charge
+type PaymentSource struct {
+	Type            string `json:"object"`
+	ID              string `json:"id"`
+	Card            *Card
```

Comment:
err, as in the following JSON blob will deserialize in an unexpected way:

```
{
    "type": "card",
    "card": {
        "number": "12345"
    },
    "expiry": "1/23"
}
```

(a similar thing will be true for the case of JSON marshaling)